src/modules/csm/distribution.py

```diff
@@ -0,0 +1,293 @@
+            log_operator.performance_coefficients = curve_params.perf_coeffs
+
+            active_validators.sort(key=lambda v: v.index)
+            numbered_validators = enumerate(active_validators, 1)
+            for key_number, validator in numbered_validators:
+                key_threshold = max(network_perf - curve_params.perf_leeway_data.get_for(key_number), 0)
+                key_reward_share = curve_params.reward_share_data.get_for(key_number)
+
+                duties = self.state.get_validator_duties(frame, validator.index)
+
+                validator_duties_outcome = self.get_validator_duties_outcome(
+                    validator,
+                    duties,
+                    key_threshold,
+                    key_reward_share,
+                    curve_params.perf_coeffs,
+                    log_operator,
+                )
+                if validator_duties_outcome.strikes:
+                    frame_strikes[(no_id, validator.pubkey)] = validator_duties_outcome.strikes
+                    log_operator.validators[validator.index].strikes = validator_duties_outcome.strikes
+                participation_shares[no_id] += validator_duties_outcome.participation_share
+                total_rebate_share += validator_duties_outcome.rebate_share
+
+        rewards_distribution_map = self.calc_rewards_distribution_in_frame(
+            participation_shares, total_rebate_share, rewards_to_distribute
+        )
+        distributed_rewards = sum(rewards_distribution_map.values())
+        # All rewards to distribute should not be rebated if no duties were assigned to validators or
+        # all validators were below threshold.
+        rebate_to_protocol = 0 if not distributed_rewards else rewards_to_distribute - distributed_rewards
+
+        for no_id, no_rewards in rewards_distribution_map.items():
+            log.operators[no_id].distributed = no_rewards
+        log.distributable = rewards_to_distribute
+        log.distributed_rewards = distributed_rewards
+        log.rebate_to_protocol = rebate_to_protocol
+
+        return rewards_distribution_map, distributed_rewards, rebate_to_protocol, frame_strikes
+
+    def _get_network_performance(self, frame: Frame) -> float:
+        att_aggr = self.state.get_att_network_aggr(frame)
+        prop_aggr = self.state.get_prop_network_aggr(frame)
+        sync_aggr = self.state.get_sync_network_aggr(frame)
+        network_perf = PerformanceCoefficients().calc_performance(ValidatorDuties(att_aggr, prop_aggr, sync_aggr))
+        return network_perf
+
+    @staticmethod
+    def get_validator_duties_outcome(
+        validator: LidoValidator,
+        duties: ValidatorDuties,
+        threshold: float,
+        reward_share: float,
+        perf_coeffs: PerformanceCoefficients,
+        log_operator: OperatorFrameSummary,
+    ) -> ValidatorDutiesOutcome:
+        if duties.attestation is None or duties.attestation.assigned == 0:
```

Comment:
add some note about the case when you have no assigned attestations, but you're in sync committee